pkg/processor/eventprocessor/pool.go

```diff
@@ -115,6 +115,11 @@ func (sa *syncPoolAllocator) GetNumObjectsAvailable() int {
 
 func (sa *syncPoolAllocator) SetObjects(objects []EventProcessor) error {
 	sa.objects = objects
+	sa.objectsChan = make(chan EventProcessor, len(objects))
+	for _, object := range objects {
+		sa.objectsChan <- object
+	}
```

Comment:
Is there a change we had object in the channel beforehand?
What happens to them in that case?

Reply:
@TomerShor there is no such chance now, I didn't want to interfere another change in process (for dynamic pool), so this will be added in upcoming pr.